app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -45,10 +45,8 @@ def authenticate_user
 
   def handle_second_factor_locked_user(type)
     analytics.track_event(Analytics::MULTI_FACTOR_AUTH_MAX_ATTEMPTS)
-
-    sign_out
-
     render 'two_factor_authentication/shared/max_login_attempts_reached', locals: { type: type }
+    sign_out
```

Comment:
As for the nil error on current_user.decorate, I would solve that by capturing the necessary view info before signing the user out, then passing that data to the view via locals. The user itself does not need to be present when the view is rendered. We just need to be able to display the remaining time before lockout expires, which is based on the `second_factor_locked_at` user attribute, which can be captured before the user signs out and is not going to change after the user is signed out.